File: lib/src/main/java/com/sha/formvalidator/validator/NumericValidator.java
package com.sha.formvalidator.validator;

import android.text.TextUtils;
import android.widget.EditText;

/**
 * A validator that returns true only if the input field contains only numbers.
 *
 */
public class NumericValidator extends Validator {

    public NumericValidator(String errorMessage) {
        super(errorMessage);
    }

    public boolean isValid(EditText et) {
        return TextUtils.isDigitsOnly(et.getText());
    }
}

File: lib/src/main/java/com/sha/formvalidator/validator/NumericValidator.kt
package com.sha.formvalidator.validator

import android.text.TextUtils
import android.widget.EditText

/**
 * A validator that returns true only if the input field contains only numbers.
 *
 */
class NumericValidator(errorMessage: String) : Validator(errorMessage) {

    override fun isValid(et: EditText): Boolean {
        return TextUtils.isDigitsOnly(et.text)
    }
}
